File: angular/projects/site/src/lib/field-types/table/table-row.ts
/** One row - what a `Table` field's value is a list of. Mirrors the server's `TableRow`. Unlike
 * `MatrixBlockValue` there is no type-tag: every row shares the same `Table.Columns` schema. */
export interface TableRow {
  values: Record<string, unknown>;
}

/** Reads a stored Table field's value - a list of rows - defensively. */
export function normalizeTableRows(source: unknown): TableRow[] {
  if (!Array.isArray(source)) {
    return [];
  }

  return source.map(item => {
    const value = (item ?? {}) as Partial<TableRow>;
    return { values: (value.values as Record<string, unknown>) ?? {} };
  });
}

File: src/Dignite.FlexFields.Site/Dignite/FlexFields/Site/Table/TableRow.cs
using Dignite.Abp.FlexFields;

namespace Dignite.FlexFields.Site.Table;

/// <summary>
/// One row - what a <c>Table</c> field's value is a list of. Unlike <c>MatrixBlockValue</c> there is no
/// type-tag: every row shares the same <see cref="TableConfiguration.Columns"/> schema, so there is
/// nothing to disambiguate. Reuses <see cref="FlexFieldDictionary"/> as-is for the cell values, same as
/// <c>MatrixBlockValue.Values</c>.
/// </summary>
public class TableRow
{
    public FlexFieldDictionary Values { get; set; } = new();
}
